core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/Controller.scala

```diff
@@ -217,7 +217,7 @@ class Controller(
     outputMessages.foreach(transferService.send)
     cp.asyncRPCClient.sendToClient(
       ExecutionStatsUpdate(
-        cp.workflowExecution.getRunningRegionExecutions.flatMap(_.getStats).toMap
+        cp.workflowExecution.getAllRegionExecutionsStats
```

Comment:
what is the reason for this change? I am not saying that you are wrong, but I want to understand why do we need this change, what is the motivation?

Reply:
The main reason for this is the `Join` operator. If we only aggregate statistics from running regions, we cannot aggregate the "build" operator's stat from the first region when the second region is running.

Reply:
we should not update "build operator"'s stats while the probe (second) region is running.

But I get your point in this case. the backend is reporting partial stats to frontend and frontend may not have the logic to aggregate. We can keep this case this way.